packages/client/src/editor/components/StickyNote.tsx

```diff
@@ -1,79 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import EditIcon from '@material-ui/icons/Edit';
 import DeleteIcon from '@material-ui/icons/Delete';
 import { UUID } from '../../types';
 
 export interface NoteProps {
-    id: UUID;
-    get: (id: UUID) => string;
-    save: (content: string, id: UUID) => void;
-    delete: (id: UUID) => void;
+  id: UUID;
+  get: (id: UUID) => string;
+  save: (content: string, id: UUID) => void;
+  delete: (id: UUID) => void;
 }
 interface NoteState {
-    editing: boolean;
-    _newText: string
+  editing: boolean;
+  _newText: string;
 }
 
 const noteStyle = {
-    width: '200px',
-    height: '100px',
-    backgroundColor: '#f2d233',
-    transform: 'translate(400px,400px)'
-}
+  width: '200px',
+  height: '100px',
+  backgroundColor: '#f2d233',
+  transform: 'translate(400px,400px)'
+};
 
-class StickyNote extends React.Component<NoteProps, NoteState> {
-    constructor(props: NoteProps) {
-        super(props);
-        this.state = {
-            editing: false,
-            _newText: '',
-        }
-        this.save = this.save.bind(this);
-        this.edit = this.edit.bind(this);
-        this.delete = this.delete.bind(this);
-        this.renderForm = this.renderForm.bind(this);
-        this.renderDisplay = this.renderDisplay.bind(this);
-    }
+const StickyNote: React.FunctionComponent<NoteProps> = props => {
+  const [isEditing, setIsEditing] = useState(false);
 
-    edit() {
-        this.setState({editing: true});
-    }
+  // TODO this won't recevie update correctly when the note text
+  // changes not because of this form, but becaus of an update from server
+  const [newText, setNewText] = useState(props.get(props.id));
 
-    save(event: React.FormEvent<HTMLFormElement>) {
-        this.setState({_newText: event.currentTarget.value})
```

Comment:
this didn't work, because the target even 'submit' event is the `form`, and not the `input` which would have the value we want